client/go/src/github.com/vespa-engine/vespa/utils/http.go

```diff
@@ -0,0 +1,56 @@
+// Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+// A HTTP wrapper which handles some errors and provides a way to replace the HTTP client by a mock.
+// Author: bratseth
+
+package utils
+
+import (
+    "net/http"
+    "net/url"
+    "strings"
+    "time"
+)
+
+// Set this to a mock HttpClient instead to unit test HTTP requests
+var ActiveHttpClient = CreateClient(time.Second * 10)
+
+type HttpClient interface {
+    Do(request *http.Request, timeout time.Duration) (response *http.Response, error error)
+}
+
+type defaultHttpClient struct {
+    client http.Client
+}
+
+func (c defaultHttpClient) Do(request *http.Request, timeout time.Duration) (response *http.Response, error error) {
```

Comment:
A copy of `c` is passed to this method. You probably want `c *defaultHttpClient`.

Reply:
Right, thanks.